src/UnicycleTrajectoryGenerator.cpp

```diff
@@ -7,6 +7,20 @@
 
 #include "UnicycleTrajectoryGenerator.h"
 
+bool UnicycleTrajectoryGenerator::clearAndAddMeasuredStep(std::shared_ptr<FootPrint> foot, Step &previousStep, const iDynTree::Vector2 &measuredPosition, double measuredAngle)
+{
+    foot->clearSteps();
+
+    Step correctedStep = previousStep;
+
+    correctedStep.position = measuredPosition;
+    iDynTree::Rotation initialRotation = iDynTree::Rotation::RotZ(previousStep.angle);
+    iDynTree::Rotation measuredRotation = iDynTree::Rotation::RotZ(measuredAngle);
+    correctedStep.angle = previousStep.angle + (initialRotation.inverse()*measuredRotation).asRPY()(2);
```

Comment:
I don't understand how you evaluate the `correctedStep.angle`.
Indeed with this  `(initialRotation.inverse()*measuredRotation).asRPY()(2)` you evaluate the angular displacement between the previous step and the actual step and then you add this angular displacement to the previous step. So why don't you use directly the `measuredAngle` variable?  
Last but not least the result of the summation of two angles should be wrapped (e.g +- PI)

Reply:
Yes you are right. They should be wrapped and actually the angle you get from iDynTree is wrapped. The problem is that that angle is integrated and I wanted to avoid jumps in a state variable. In addition, when I tried to plug directly the wrapped angle on the planner, I saw the robot rotating the leg of 180deg around the yaw axis. This was due to the interpolator. If the previous step was having a yaw angle of 179 deg and the following one is -175deg, because of the interpolator, you will have the leg rotating of 174deg in the wrong direction. In order to avoid both problems I stuck to the un-wrapped version, using the simple trick above which  
> evaluate the angular displacement between the previous step and the actual step and then [...] add this angular displacement to the previous step